File: julia/problem_09.jl
# Copyright 2014 Mitchell Kember. Subject to the MIT License.
# Project Euler: Problem 9
# Special Pythagorean triplet

module problem_09

function main()
	for a = 1:332, b=a+2:div(1000-a-1, 2)
		c = 1000 - a - b
		if a^2 + b^2 == c^2
			return a * b * c
		end
	end
end

end # module

File: haskell/Problem09.hs
-- Copyright 2015 Mitchell Kember. Subject to the MIT License.
-- Project Euler: Problem 9
-- Special Pythagorean triplet

module Problem09 where

import Common (isPythagorean)

import Data.List (find)
import Data.Maybe (fromJust)

solve :: Int
solve = mult . fromJust . find isPythagorean $ triplets
  where
    triplets = [(a, b, 1000-a-b) | a <- [1..332], b <- bSource a]
    bSource a = [a + 1 .. (1000 - a) `div` 2 - 1]
    mult (a, b, c) = a * b * c
